### PR TITLE
Feature/mbapi

### PR DESCRIPTION
**Pull request by frwg**
1. The key pair is generated using `bin/console lexik:jwt:generate-keypair`, with keys stored in `application/config/jwt`. This directory is excluded from version control via `.gitignore` and is integrated into the `bootstrap` command.

2. When merging, ensure the dependent branch on `mapbender/mapbender` is merged first, then update `composer.json` and `composer.lock` to reflect any dependency changes.

3. The JWT token can be obtained by sending authentication data (in JSON format: `{ "username": "<username>", "password": "<password>" }`) to `<server_url>/api/login_check`. 
 
Example Curl command: 
```bash
curl -X POST http://mapbender.localhost/api/login_check  -H "Content-Type: application/json"  -d '{"username": "<username>", "password": "<password>"}'`
```

4. Extend your virtual host configuration according to https://github.com/lexik/LexikJWTAuthenticationBundle/blob/3.x/Resources/doc/index.rst#important-note-for-apache-users

5. The class responsible for handling API requests is located in `mapbender/mapbender` at `application/mapbender/src/Mapbender/CoreBundle/Controller/ApiController.php`. Access to endpoints such as `<server_url>/api/example` or `<server_url>/api/my_custom_command` is granted through this authentication token.

Example curl command:

```
curl -X GET -H "Authorization: Bearer <token>" http://mapbender.localhost/api/example
```

### CODE REVIEW

File: application/config/packages/doctrine.yaml [hunk 1]
```diff
@@ -4,17 +4,12 @@ doctrine:
         connections:
             default:
                 url: '%env(resolve:MAPBENDER_DATABASE_URL)%'
+            geodata_db:
```
Comment (Phocacius) [inline application/config/packages/doctrine.yaml:7]: please don't commit additional databases to the core repo

File: application/.env [hunk 2]
```diff
@@ -37,3 +37,9 @@ MAPBENDER_DATABASE_URL="sqlite:///%kernel.project_dir%/var/db/demo.sqlite"
 ###> mapbender/mapbender ###
 KERNEL_CLASS="\Mapbender\BaseKernel"
 ###< mapbender/mapbender ###
+
+###> lexik/jwt-authentication-bundle ###
+JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
+JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
+JWT_PASSPHRASE=b62cfe7b2ee8582654e0d2a6a677a0014806194640a6b93ff359cbd5452c2cf2
```
Comment (Phocacius) [inline application/.env:44]: I don't think this should be committed. I suggest to have a placeholder here (something like "change me") and than have an additional command in the bootstrap script that auto-generates a passphrase and places it in the .env.local file if it does not yet exist

File: application/symfony.lock [hunk 3]
```diff
@@ -25,6 +25,18 @@
             "src/Repository/.gitignore"
         ]
     },
+    "lexik/jwt-authentication-bundle": {
+        "version": "2.21",
```
Comment (Phocacius) [inline application/symfony.lock:29]: why did you use an old version here? It's so old it does not support PHP 8.3:

```
  Problem 1
    - lcobucci/clock is locked to version 3.0.0 and an update of this package was not requested.
    - lcobucci/clock 3.0.0 requires php ~8.1.0 || ~8.2.0 -> your php version (8.3.13) does not satisfy that requirement.
```

File: application/config/packages/security.yaml [hunk 4]
```diff
@@ -15,6 +15,20 @@ security:
         login:
             pattern:  ^/user/login$
             security: false
+
+        jwt_login:
+          pattern: ^/api/login_check
```
Comment (Phocacius) [inline application/config/packages/security.yaml:20]: the login check creates a html symfony message if the user does not exist, this should not happen in an API and rather return an appropriate error code:

grafik

**Review by Phocacius — CHANGES_REQUESTED** [review]

File: application/config/packages/security.yaml (hunk 4 above)
Comment (frwg) [reply application/config/packages/security.yaml:20]: What should the response look like?  Simple JSON containing an error code?

Comment (Phocacius) [reply application/config/packages/security.yaml:20]: Yes, HTTP Code 401 should remain and IMO as body something like `{"error": "User <username> not found"}` is sufficient.

Comment (Phocacius) [reply application/config/packages/security.yaml:20]: fixed

**Pull request merged**